### PR TITLE
[ci] Add validation_helper_spec.rb

### PR DESCRIPTION
**Pull request by mschnitzer**
This implements the missing tests for the rspec test suite.

Related to card:
https://trello.com/c/CJtlSMoy/268-app-helpers-validation-helper-rb

### CODE REVIEW

File: src/api/spec/helpers/validation_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  describe '#valid_project_name?' do
+    it 'calls Project.valid_name?' do
+      expect(Project).to receive(:valid_name?).with("home:mschnitzer")
+      valid_project_name?("home:mschnitzer")
+    end
+  end
+
+  describe '#valid_project_name!' do
+    it 'raises an exception if a project name is invalid' do
+      expect { valid_project_name!("home mschnitzer") }.to raise_error(ValidationHelper::InvalidProjectNameError)
```
Comment (Ana06) [inline src/api/spec/helpers/validation_helper_spec.rb]: good place to use property tests.

**Comment by bgeuken** [comment]
LGTM

File: src/api/spec/helpers/validation_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    end
+
+    it 'does not raise an exception on valid project names' do
+      expect(valid_project_name!("home:mschnitzer")).to eq(nil)
```
Comment (Ana06) [inline src/api/spec/helpers/validation_helper_spec.rb]: why don't you test what you say in the message? That it doesn't raise an exception:

`expect { valid_project_name!("home mschnitzer") }.to_not raise_error(ValidationHelper::InvalidProjectNameError)`

File: src/api/spec/helpers/validation_helper_spec.rb (hunk 1 above)
Comment (mschnitzer) [inline src/api/spec/helpers/validation_helper_spec.rb]: Good point, will do ;-)

Comment (mschnitzer) [inline src/api/spec/helpers/validation_helper_spec.rb]: In the end it doesn't make sense to test all situations where project name can be invalid. The RegEx would be to complex I guess. Besides that, it's not the functions task to test all scenarios of valid project names. For that we've https://github.com/openSUSE/open-build-service/blob/master/src/api/spec/models/project_spec.rb#L100

**Comment by mschnitzer** [comment]
@Ana06 please review it again :-)

File: src/api/spec/helpers/validation_helper_spec.rb (hunk 1 above)
Comment (Ana06) [inline src/api/spec/helpers/validation_helper_spec.rb]: of course that it makes sense, if you are only testing that it raises an exception with `home mschnitzer`, not when name is invalid.

And yesterday you liked the idea... you removed the comment... :unamused: 

Anyway, I am okey with merging this. I can add property tests afterwards if you don't want to do it.

**Pull request merged**